datascience/maps.py

```diff
@@ -0,0 +1,510 @@
+            setattr(self, k, v)
+        return self
+    
+    def get(self, *args):
+        """ Get a set of variables - use pairs to set getattr default """
+        get = lambda val: getattr(self, val[0], val[1]) \
+            if isinstance(val, tuple) else getattr(self, val)
+        return (get(var) for var in args)
+
+    @_hook
+    def map(self, map=None):
+        """ Maps this object, with its own attributes """
+        if not callable(self._mapper):
+            self._mapper = getattr(map, self._mapper)
+        self._folium = self._mapper(**self._attributes)
+        return self
+
+
+class Map(MapEntity):
+    """
+    Represents a map, potentially with MapPoints and MapRegions
+    - For description of all capabilities, see Folium Map:
+      https://github.com/python-visualization/folium/blob/master/folium/folium.py#L58
+      
+    points -- a list of MapPoints
+    regions -- a list of MapRegions
+
+    >>> map = Map().map()
+    >>> map.to_html()
+    <IPython.core.display.HTML object>
+    """
+
+    _mapper = folium.Map
+    
+    _allowed_attributes = [
+        'location',     # location or center -- lat-long pair at the center of the map
+        'width',        # width -- pixel int or percentage string
+        'height',       # height -- pixel int or percentage string
+        'tiles',        # tiles -- map tileset, or "theme"
+        'API_key',      # API_key -- for Cloudmade or Mapbox tiles
+        'max_zoom',     # max_zoom -- maximum zoom, default 18
+        'zoom_start',   # zoom_start -- starting zoom level
+    ]
+
+    _defaults = {
+        'location': [45.5244, -122.6699],
+        'tiles': 'Stamen Toner',
+        'zoom_start': 17,
+        'width': 960,
+        'height': 500,
+        'points': [],
+        'regions': []
+    }
+
+    @_hook
+    def to_html(self):
+        """
+        Takes in a folium map as a parameter and outputs the HTML that
+        IPython will display.
+        """
+        map = self._folium
+        map._build_map()
+        map_html = map.HTML.replace('"', '&quot;')
+        return HTML('<iframe srcdoc="%s" '
+                    'style="width: %spx; height: %spx; '
+                    'border: none"></iframe>' % (map_html, map.width, map.height))
+    
+    def _before_to_html(self, *args, **kwargs):
+        """
+        Called before to_html, add points and regions before outputting map 
+        """
+        for point in self.points:
+            point.map(self._folium)
+
+        for region in self.regions:
+            region.map(self._folium)
+
+        # uncomment the following with new Folium release
+        # if 'zoom_start' not in self._init:
+        #     self._autofit(self.bounds)
+    
+    def _before_map(self, *args, **kwargs):
+        """ Autozoom map where appropriate """
+        self.bounds = self._autobounds()
+        
+        if not self.location:
+            self._autocenter(self.bounds)
+
+        # remove with new Folium release
```

Comment:
nit: Maybe you can leave your name here so a future dev knows who to ask about this? I leave comments like `TODO(sam): do stuff` for this purpose.